subsys/storage/flash_map/flash_map.c

```diff
@@ -16,7 +16,6 @@
 #include <zephyr/storage/flash_map.h>
 #include "flash_map_priv.h"
 #include <zephyr/drivers/flash.h>
-#include <soc.h>
```

Comment:
Do we actually need changes to Zephyr mainline kernel? If so, at least it should be added in separate commit with proper explanation, I don't understand the reason of such change. It may break other SOC.

Reply:
I don't know why this line present in those files. There are no dependency from soc.h.
Technically we can create empty soc.h for our board but why?

Reply:
I see only 2 valid approaches:
1. Create empty soc.h header to comply mainline code
2. Remove this includes from subsys with proper explanation and try to send it to mainline repo
Any of these changes should be in separate commit with explanation in commit message.

Reply:
I believe this is an error of the author. Went through all of the soc.h files and see the following:
1) soc,h for different platforms have completely different definitions. Nothing common at all
2) Not all platforms have soc.h files which means this header is optional.
I think that common code is not a correct place for optional header without common interface.
I suggest to contact an author Andrzej Puzdrowski and ask him specifically. Who knows, maybe he will accept this change fast.